File: pinyougou-interface/src/main/java/com/pinyougou/core/service/BrandService.java
package com.pinyougou.core.service;

import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 查询所有品牌
     * @return
     */
    public List<Brand> findAll();

    /**
     * 品牌的分页查询
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(Integer pageNum,Integer pageSize);

    /**
     * 品牌的条件查询
     * @param pageNum
     * @param pageSize
     * @param brand
     * @return
     */
    public PageResult search(Integer pageNum,Integer pageSize,Brand brand);

    /**
     * 新增品牌
     * @param brand
     */
    public void add(Brand brand);

    /**
     * 品牌回显(根据id获取实体)
     */
    public Brand findOne(Long id);

    /**
     * 修改品牌
     * @param brand
     */
    public void update(Brand brand);

    /**
     * 品牌的批量删除
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 新增模板时初始化品牌列表
     * @return
     */
    public List<Map<String,String>> selectOptionList();

    /**
     * 更新品牌状态
     * @param ids
     * @param status
     */
    void updateStatus(Long[] ids, String status);
}

File: pinyougou-web-manager/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	
	this.findPage = function(page,rows){
		return $http.get("../brand/findByPage.do?pageNum="+page+"&pageSize="+rows);
	}
	
	this.add = function(entity){
		return $http.post("../brand/add.do",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	this.findOne=function(id){
		return $http.get("../brand/findById.do?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/delete.do?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../brand/search.do?pageNum="+page+"&pageSize="+rows,searchEntity);
	}
	
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.do");
	}

    this.updateStatus = function(ids,status){
        return $http.get('../brand/updateStatus.do?ids='+ids+"&status="+status);
    }
});